frontend/src/components/widgets/DataGrid/DataGrid.tsx

```diff
@@ -180,10 +180,47 @@ function DataGrid({
     onColumnResized,
   } = useDataLoader(element)
 
-  const [tableWidth, setTableWidth] = useState(propWidth)
+  const [width, setWidth] = useState(propWidth)
 
   const dataEditorRef = React.useRef<DataEditorRef>(null)
 
+  useLayoutEffect(() => {
+    // Without this timeout, the canvas does not reflect the actual size sometimes
```

Comment:
nit: May be useful to note in this comment that the timeout helps because it ensures that the execution of this function is placed after the component render in the event loop

Reply:
thanks, that's a good addition!